ghcide/src/Development/IDE/Plugin/TypeLenses.hs

```diff
@@ -305,26 +323,40 @@ rules recorder = do
     result <- liftIO $ gblBindingType (hscEnv <$> hsc) (tmrTypechecked <$> tmr)
     pure ([], result)
 
+bindToSig :: Id -> HscEnv -> GlobalRdrEnv -> IOEnv (Env TcGblEnv TcLclEnv) (Name, String)
+bindToSig id hsc rdrEnv = do
+    env <-
+#if MIN_VERSION_ghc(9,7,0)
+      liftZonkM
+#endif
+      tcInitTidyEnv
```

Comment:
https://github.com/haskell/haskell-language-server/pull/2966#discussion_r900008476

> is this expensive? It seems like we might do it a lot...

Reply:
Reading the GHC comment, 

> We initialise the "tidy-env", used for tidying types before printing,
> by building a reverse map from the in-scope type variables to the
> OccName that the programmer originally used for them

It *sounds* at least expensive. However, it is now cached by a rule, so should be fine.